sr5824-msr9732-sm9453/project.md

```diff
@@ -0,0 +1,136 @@
+---
+title: "Interoperability in the Heterogeneous Cloud Environment"
+author:
+  - Sriram Ramesh <sriram.ramesh@nyu.edu>
+  - Metarya Ruparel <msr9732@nyu.edu>
+  - Safwan Mahmood <sm9453@nyu.edu>
+---
+
+For our Big Data & Machine Learning final project we plan to hypothesize a solution
+for **Interoperability in the Heterogeneous Cloud Environment**, a blatant issue 
+industry wide.
+
+# Motivation
+
+After the burst of cloud platforms in 2006 with the introduction of Amazon
+Web Services, a lot of companies started migrating their services to the 
+cloud platform from a bare metal lying in one of their own datacenters.
+Big hulking datacenters are a necessary evil, a barely concealed secret. 
+No one wants to build one. They’re certainly impressive – they resemble the 
+set of a science fiction movie, row upon row of silent machines. But they’re 
+wildly expensive, and their maintenance challenges are never ending. 
+
+Cloud Platforms provided these companies the ability to access shared, online computing 
+resources, without the over head of maintaining them. However, these providers often 
+offer their own proprietary applications, interfaces, APIs and infrastructures, which
+somewhat makes these providers unique. Due to the uniqueness (be it performance or costing)
+companies would want to onboard different set of services onto various providers.
+For example, Salesforce.com, which  started the wave of SaaS back in the late 1900's 
+, has only recently started it's migration from 1st party datacenters to these 3rd party
+providers. Some of their micro-services are being on-boarded to GCP, while a couple of
+others to AWS, resulting in a heterogeneous cloud environment. This heterogeneous 
+environment makes it difficult for companies to change cloud service providers; in 
+any case. Exploring capabilities to support the automated migration from one provider 
+to another is an active, open research area. A lot of people have been pursuing approaches 
+to reduce the impact of vendor lock-in by investigating the cloud migration problem 
+at the level of the VM. However, the migration downtime, decoupling VM from
+underlying systems and security of live channels remain open issues and as a part of
+this project we aim to hypothesize a solution to at least one blatant issue in the
+process of migration.
```

Comment:
It would be really good to see some links or evidence of this solution, mostly because the current description lends itself to various solutions:
* You could be talking about what Kubernetes allegedly does: provide a single interface to resources.
* You could be talking about something like Pulumi (https://www.pulumi.com/) or Terraform (https://www.terraform.io/).
* You could be talking about a third thing.

If as you say there are lots of people working on this then you should cite them.

Reply:
We are trying to have Interoperability between the cloud providers by
replicating cloud specific solutions like AWS batch jobs and reddis (https://core.ac.uk/download/pdf/46571839.pdf).
IBM Multicloud provides an interface to kubernetes of each cloud provider. We want to supplement that solution by providing static code analyser to migrate existing solutions to any cloud. 
Our idea is to code for one cloud and deploy it across clouds irrespective of the cloud specific solutions.